CodeEdit/Features/Tabs/TabGroup/TabGroupData.swift

```diff
@@ -8,25 +8,28 @@
 import Foundation
 import OrderedCollections
 import DequeModule
+import SwiftUI
 
 final class TabGroupData: ObservableObject, Identifiable {
     typealias Tab = WorkspaceClient.FileItem
 
     /// Set of open tabs.
     @Published var tabs: OrderedSet<Tab> = [] {
         didSet {
-            let change = tabs.symmetricDifference(oldValue)
-
-            if tabs.count > oldValue.count {
-                // Amount of tabs grew, so set the first new as selected.
-                selected = change.first
-            } else {
-                // Selected file was removed
-                if let selected, change.contains(selected) {
-                    if let oldIndex = oldValue.firstIndex(of: selected), oldIndex - 1 < tabs.count, !tabs.isEmpty {
-                        self.selected = tabs[max(0, oldIndex-1)]
-                    } else {
-                        self.selected = nil
+            withAnimation(.linear(duration: 0)) {
```

Comment:
Please don't use `withAnimation` in a ViewModel. Instead, try using animation modifiers in the view itself, so view logic and data logic stay separated